packages/host/app/components/operator-mode/code-submode/spec-preview-badge.gts

```diff
@@ -0,0 +1,103 @@
+import { on } from '@ember/modifier';
+
+import Component from '@glimmer/component';
+
+import Check from '@cardstack/boxel-icons/check';
+
+import { BoxelButton } from '@cardstack/boxel-ui/components';
+import { IconPlus } from '@cardstack/boxel-ui/icons';
+
+import { Spec, type SpecType } from 'https://cardstack.com/base/spec';
+
+interface SpecPreviewBadgeSignature {
+  Args: {
+    spec?: Spec;
+    numberOfInstances?: number;
+    showCreateSpec: boolean;
+    createSpec: (event: MouseEvent) => void;
+    isCreateSpecInstanceRunning: boolean;
+  };
+}
+
+export default class SpecPreviewBadge extends Component<SpecPreviewBadgeSignature> {
+  private get moreThanOneInstance() {
+    return this.args.numberOfInstances && this.args.numberOfInstances > 1;
+  }
+
+  private get specType() {
+    return this.args.spec?.specType as SpecType | undefined;
+  }
+
+  <template>
+    <span class='spec-indicator'>
+      {{#if @showCreateSpec}}
+        <BoxelButton
+          class='create-spec-button'
+          @icon='plus'
+          @kind='primary'
+          @size='extra-small'
+          @loading={{@isCreateSpecInstanceRunning}}
+          {{on 'click' @createSpec}}
+          data-test-create-spec-button
+        >
+          {{#unless @isCreateSpecInstanceRunning}}
+            <IconPlus width='10px' height='10px' />
+          {{/unless}}
+        </BoxelButton>
+      {{else if this.moreThanOneInstance}}
+        <Check
+          class='spec-checkmark'
+          data-test-number-of-instance={{@numberOfInstances}}
+          title='{{@numberOfInstances}} instances'
+        />
+      {{else}}
+        {{#if this.specType}}
+          <Check
+            class='spec-checkmark'
+            data-test-spec-tag={{this.specType}}
```

Comment:
Perhaps this is not needed anymore

I think inside test I usually check this already

```
    assert.dom('[data-test-exported-type]').containsText('card');
```